smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/HttpProtocolUnitTestResponseGenerator.kt

```diff
@@ -147,18 +150,27 @@ open class HttpProtocolUnitTestResponseGenerator protected constructor(builder:
 fun renderMemberAssertions(writer: SwiftWriter, test: HttpMessageTestCase, members: Collection<MemberShape>, model: Model, symbolProvider: SymbolProvider, expected: String, actual: String) {
     for (member in members) {
         val shape = model.expectShape(member.target.toShapeId())
-        val expectedMemberName = "$expected.${symbolProvider.toMemberName(member)}"
-        val actualMemberName = "$actual.${symbolProvider.toMemberName(member)}"
+        val baseVarName = symbolProvider.toMemberName(member)
+        val expectedMemberName = "$expected.$baseVarName"
+        val actualMemberName = "$actual.$baseVarName"
+        val suffix = if (symbolProvider.toSymbol(member).isBoxed()) "?" else ""
         if (member.isStructureShape) {
             writer.write("XCTAssert(\$L === \$L)", expectedMemberName, actualMemberName)
         } else if ((shape.isDoubleShape || shape.isFloatShape)) {
             val stringNodes = test.params.stringMap.values.map { it.asStringNode().getOrNull() }
             if (stringNodes.isNotEmpty() && stringNodes.mapNotNull { it?.value }.contains("NaN")) {
-                val suffix = if (symbolProvider.toSymbol(member).isBoxed()) "?" else ""
                 writer.write("XCTAssertEqual(\$L$suffix.isNaN, \$L$suffix.isNaN)", expectedMemberName, actualMemberName)
             } else {
                 writer.write("XCTAssertEqual(\$L, \$L)", expectedMemberName, actualMemberName)
             }
+        } else if (shape.isBlobShape && shape.hasTrait<StreamingTrait>()) {
```

Comment:
This renders an `XCAssert` for comparing two data streams.  It gets the data to local variables then compares them.

The final result, as rendered to a protocol test:
```
        // Compare blobs by reading them both to data
        let expectedBlobData = try await expected.blob?.readData()
        let actualBlobData = try await actual.blob?.readData()
        XCTAssertEqual(expectedBlobData, actualBlobData)
```

(btw `blob` is the Smithy equivalent of Swift `Data`.)